### PR TITLE
[CPU][DEBUG]Fix "getData" throw exception issue in print of debug log.

### PR DESCRIPTION
**Pull request by xipingyan**
### Details:
 - *`getData` maybe throw exception after [PR21642](https://github.com/openvinotoolkit/openvino/pull/21642) is merged, when dumping debug log.*
 - *Catch exception and print ?*

### CODE REVIEW

**Review by luo-cheng2021 — APPROVED** [review]
LGTM

File: src/plugins/intel_cpu/src/utils/debug_capabilities.cpp [hunk 1]
```diff
@@ -198,8 +198,7 @@ std::ostream & operator<<(std::ostream & os, const Node &c_node) {
                     replace_all(shape_str, " ", "");
                     leftside << comma << desc->getPrecision().get_type_name()
                                 << "_" << desc->serializeFormat()
-                                << "_" << shape_str
-                                << "_" << ptr->getData();
```
Comment (ceciliapeng2011) [inline src/plugins/intel_cpu/src/utils/debug_capabilities.cpp:202]: suggest keep it and work around this way like -
<< "_" << ptr->getSize() > 0 ? ptr->getData() : "invalid"

Comment (xipingyan) [reply src/plugins/intel_cpu/src/utils/debug_capabilities.cpp:202]: I will try.

Comment (xipingyan) [reply src/plugins/intel_cpu/src/utils/debug_capabilities.cpp:202]: Updated.
1: When we call `getSize()` here, sometimes trigger new issue:
`Can't get memory size for undefined shape`
2: `typeid().name` only can current variable type, for example: we only get based class type here.
3: I add `hasActualMemory()` to judge if we can call `getData()`;

Comment (ceciliapeng2011) [reply src/plugins/intel_cpu/src/utils/debug_capabilities.cpp:202]: I prefer #1.  We can check if the shape is static first.

Comment (xipingyan) [reply src/plugins/intel_cpu/src/utils/debug_capabilities.cpp:202]: As you said, `getData` is essential, if the shape is static, don't we print `getData`? @ceciliapeng2011

Comment (xipingyan) [reply src/plugins/intel_cpu/src/utils/debug_capabilities.cpp:202]: Catch this exception, and then print "?", it seems reasonable and it has smaller changes. @ceciliapeng2011 is it OK?

File: src/plugins/intel_cpu/src/utils/debug_capabilities.cpp [hunk 2]
```diff
@@ -184,6 +184,18 @@ std::ostream & operator<<(std::ostream & os, const Node &c_node) {
             num_output_port = edge->getInputNum() + 1;
     }
 
+    auto getData = [](MemoryPtr& ptr) {
```
Comment (ceciliapeng2011) [inline src/plugins/intel_cpu/src/utils/debug_capabilities.cpp:187]:
```suggestion
    auto getData = [](const MemoryPtr& ptr) {
```

**Review by ceciliapeng2011 — APPROVED** [review]
LGTM..

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**